pkg/resource/alias/references.go

```diff
@@ -36,17 +43,95 @@ func (rm *resourceManager) ResolveReferences(
 	apiReader client.Reader,
 	res acktypes.AWSResource,
 ) (acktypes.AWSResource, error) {
-	return res, nil
+	namespace := res.MetaObject().GetNamespace()
+	ko := rm.concreteResource(res).ko.DeepCopy()
+	err := validateReferenceFields(ko)
+	if err == nil {
+		err = resolveReferenceForFunctionName(ctx, apiReader, namespace, ko)
+	}
+
+	// If there was an error while resolving any reference, reset all the
+	// resolved values so that they do not get persisted inside etcd
+	if err != nil {
+		ko = rm.concreteResource(res).ko.DeepCopy()
+	}
+	if hasNonNilReferences(ko) {
+		return ackcondition.WithReferencesResolvedCondition(&resource{ko}, err)
+	}
+	return &resource{ko}, err
 }
 
 // validateReferenceFields validates the reference field and corresponding
 // identifier field.
 func validateReferenceFields(ko *svcapitypes.Alias) error {
+	if ko.Spec.FunctionRef != nil && ko.Spec.FunctionName != nil {
+		return ackerr.ResourceReferenceAndIDNotSupportedFor("FunctionName", "FunctionRef")
+	}
```

Comment:
Remember the case of users providing `FunctionName` and `FunctionRef`? Looks like it's handled here

Reply:
Yeah, I see.